src/item_pocket.h

```diff
@@ -0,0 +1,174 @@
+
+        item &back();
+        const item &back() const;
+        item &front();
+        const item &front() const;
+        size_t size() const;
+        void pop_back();
+
+        ret_val<contain_code> can_contain( const item &it ) const;
+
+        // combined volume of contained items
+        units::volume contains_volume() const;
+        units::volume remaining_volume() const;
+        units::volume volume_capacity() const;
+        // combined weight of contained items
+        units::mass contains_weight() const;
+        units::mass remaining_weight() const;
+
+        units::volume item_size_modifier() const;
+        units::mass item_weight_modifier() const;
+
+        item *magazine_current();
+        void casings_handle( const std::function<bool( item & )> &func );
+        bool use_amount( const itype_id &it, int &quantity, std::list<item> &used );
+        bool will_explode_in_a_fire() const;
+        bool detonate( const tripoint &p, std::vector<item> &drops );
+        bool process( const itype &type, player *carrier, const tripoint &pos, bool activate,
+                      float insulation, temperature_flag flag );
+        bool legacy_unload( player *guy, bool &changed );
+        void remove_all_ammo( Character &guy );
+        void remove_all_mods( Character &guy );
+
+        // removes and returns the item from the pocket.
+        cata::optional<item> remove_item( const item &it );
+        cata::optional<item> remove_item( const item_location &it );
+        bool spill_contents( const tripoint &pos );
+        void clear_items();
+        bool has_item( const item &it ) const;
+        item *get_item_with( const std::function<bool( const item & )> &filter );
+        void remove_items_if( const std::function<bool( item & )> &filter );
+        void has_rotten_away( const tripoint &pnt );
+
+        // tries to put an item in the pocket. returns false if failure
+        ret_val<contain_code> insert_item( const item &it );
+        void add( const item &it );
+
+        // only available to help with migration from previous usage of std::list<item>
+        std::list<item> &edit_contents();
+
+        // cost of getting an item from this pocket
+        // @TODO: make move cost vary based on other contained items
+        int obtain_cost( const item &it ) const;
+        // @relates visitable
+        VisitResponse visit_contents( const std::function<VisitResponse( item *, item * )> &func,
+                                      item *parent = nullptr );
+
+        void general_info( std::vector<iteminfo> &info, int pocket_number, bool disp_pocket_number ) const;
+        void contents_info( std::vector<iteminfo> &info, int pocket_number, bool disp_pocket_number ) const;
+
+        void load( const JsonObject &jo );
+        void serialize( JsonOut &json ) const;
+        void deserialize( JsonIn &jsin );
+
+        bool operator==( const item_pocket &rhs ) const;
+
+        bool was_loaded;
```

Comment:
Why do you need this? It is not even initialised. Same with the member of the same name in the other class.

Reply:
this member is needed to use `optional` and `mandatory` from generic_factory.h